src/server/src/test/java/io/cassandrareaper/service/SegmentRunnerTest.java

```diff
@@ -475,6 +475,145 @@ protected JmxProxy connect(
     assertEquals(2, storage.getRepairSegment(runId, segmentId).get().getFailCount());
   }
 
+  @Test
+  public void outOfOrderTest() throws InterruptedException, ReaperException, ExecutionException {
```

Comment:
verified that this test fails without the patch applied.

Reply:
can we get a variations of this test for 
 - Cassandra-2.1 and Cassandra-2.2
 - all different sequences of notifications (out of order `failed` isn't tested)

Reply:
cool